File: src/core/Scene.java
package core;

import java.awt.event.KeyEvent;
import java.awt.event.KeyListener;
import java.awt.event.MouseEvent;
import java.awt.event.MouseListener;

import javax.swing.JPanel;

public abstract class Scene extends JPanel implements KeyListener, MouseListener {
	private int cycleCount = 0;
	
	public Scene(Container container) {
		setSize(container.getDimension());
	}
	
	public int getCycleCount() {
		return cycleCount;
	}
	
	public void update() {
		cycleCount++;
	}
	
	public void start() {
		revalidate();
	}
	
	public void resume(Scene lastScene) {
		revalidate();
	}
	
	public void keyPressed(KeyEvent e) {
	}

	public void keyReleased(KeyEvent e) {
	}

	public void keyTyped(KeyEvent e) {
		
	}
	
	public void mouseClicked(MouseEvent e) {
	}

	public void mouseEntered(MouseEvent e) {
	}

	public void mouseExited(MouseEvent e) {
	}

	public void mousePressed(MouseEvent e) {
	}

	public void mouseReleased(MouseEvent e) {
	}
}

File: web/js/scene/Scene.js
export default class Scene {
    constructor(container) {
        this._width = container.getDimension().width;
        this._height = container.getDimension().height;
        this._cycleCount = 0;
    }

    getCycleCount() {
        return this._cycleCount;
    }

    update() {
        this._cycleCount++;
    }

    start() {}

    resume(lastScene) {}

    keyPressed(e) {}

    keyReleased(e) {}

    keyTyped(e) {}

    mouseReleased(e) {}

    draw(ctx) {}
}
